### PR TITLE
Add ability for plugin to flush old cache objects

### PR DESCRIPTION
**Pull request by jmccann**
Would love some feedback on how I implemented this feature.

I ran it a little and it seemed to work for me.  Was using `touch -d <dateString> <file>` to make items on my minio server "older".  Please try it out!

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -198,13 +204,20 @@ func run(c *cli.Context) error {
 		return err
 	}
 
+	flushage, err := strconv.Atoi(c.String("flush_age"))
```
Comment (donny-dont) [inline main.go]: Should be flushAge

File: main.go [hunk 2]
```diff
@@ -198,13 +204,20 @@ func run(c *cli.Context) error {
...
+	if err != nil {
+		return err
+	}
+
 	p := &Plugin{
 		Filename:     filename,
 		Path:         path,
 		FallbackPath: fallbackPath,
 		FlushPath:    flushPath,
 		Mode:         mode,
```
Comment (donny-dont) [inline main.go]: Create a new mode for Flush

File: plugin.go [hunk 3]
```diff
@@ -53,9 +56,21 @@ func (p *Plugin) Exec() error {
...
 		err = f.Flush(p.FlushPath)
 	}
 
 	return err
 }
+
+func genIsExpired(age int) cache.DirtyFunc {
+	return func(file storage.FileEntry) bool {
+		// Check if older then 30 days
+		if file.LastModified.Before(time.Now().AddDate(0, 0, age * -1)) {
```
Comment (donny-dont) [inline plugin.go]: Guessing you can just return the result of `file.LastModified.Before`

File: main.go (hunk 2 above)
Comment (donny-dont) [inline main.go]: I think this should also be usable as a command line application for controlling the cache. So I think it should be a separate mode.

**Comment by jmccann** [comment]
@donny-dont Think I got all your concerns addressed.

Also, this PR depends on https://github.com/drone/drone-cache-lib/pull/2 and https://github.com/drone/drone-cache-lib/pull/3.

**Comment by donny-dont** [comment]
K once those are merged we can merge it all.

**Pull request merged**